### ISSUE TITLE
LaTeX: automatic replacement of Unicode BOX DRAWINGS characters (and arguably others too) should be skipped with lualatex and xelatex

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#3706]
Subject: at https://github.com/sphinx-doc/sphinx/blob/master/sphinx/util/texescape.py, a few Unicode BOX DRAWINGS characters are replaced, 

```
    ('─', r'-'),
[...]
    ('╲', r'\textbackslash{}'),
[...]
    ('│', r'\textbar{}'),
```

But this prevents to incorporate output of unix tree command for example in a satisfactory way. Many OpenType fonts have these characters, hence at least for `latex_engine` set to `'lualatex'` or `'xelatex'` there should not be such replacement.

#### Procedure to reproduce the problem

``index.rst``:

```
BOX DRAWINGS LIGHT DIAGONAL UPPER LEFT TO LOWER RIGHT

╲

```
compile with `latex_engine = 'xelatex'`, then tex file contains

```
BOX DRAWINGS LIGHT DIAGONAL UPPER LEFT TO LOWER RIGHT

\textbackslash{}
```

### Environment info
- OS: mac os
- Python version: 3.5.3
- Sphinx version: 1.5.5

**Comment by jfbu** [comment]
maybe I abuse tags... I never quite know how to distinguish bug from question from enhancement...

**Comment by tk0miya** [comment]
Fixed by #3876
